internal/dao/sqlx/source_dao.go

```diff
@@ -0,0 +1,124 @@
+package sqlx
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+
+	"github.com/RHEnVision/provisioning-backend/internal/dao"
+	"github.com/RHEnVision/provisioning-backend/internal/db"
+	"github.com/RHEnVision/provisioning-backend/internal/models"
+	"github.com/jmoiron/sqlx"
+)
+
+const (
+	createSource        = `INSERT INTO sources (account_id, source_id, name, tenant_id, auth_id) VALUES ($1, $2, $3, $4, $5) RETURNING id,created_at`
+	getSourceById       = `SELECT * FROM sources WHERE id = $1;`
+	listSources         = `SELECT * FROM sources ORDER BY id LIMIT $1 OFFSET $2`
+	deleteSourceById    = `DELETE FROM sources WHERE id = $1 RETURNING source_id`
+	getSourceByTenantId = `SELECT * FROM sources WHERE tenant_id = $1 ORDER BY id LIMIT 1;`
+)
+
+type sourceDaoSqlx struct {
+	name          string
+	getById       *sqlx.Stmt
+	list          *sqlx.Stmt
+	create        *sqlx.Stmt
+	getByTenantId *sqlx.Stmt
+	deleteById    *sqlx.Stmt
+}
+
+func getSourceDao(ctx context.Context) (dao.SourceDao, error) {
+	var err error
+	daoImpl := sourceDaoSqlx{}
+	daoImpl.name = "source"
+
+	daoImpl.getById, err = db.DB.PreparexContext(ctx, getSourceById)
+	if err != nil {
+		return nil, NewPrepareStatementError(ctx, &daoImpl, getSourceById, err)
+	}
+	daoImpl.list, err = db.DB.PreparexContext(ctx, listSources)
+	if err != nil {
+		return nil, NewPrepareStatementError(ctx, &daoImpl, listSources, err)
+	}
+	daoImpl.create, err = db.DB.PreparexContext(ctx, createSource)
+	if err != nil {
+		return nil, NewPrepareStatementError(ctx, &daoImpl, createSource, err)
+	}
+	daoImpl.getByTenantId, err = db.DB.PreparexContext(ctx, getSourceByTenantId)
+	if err != nil {
+		return nil, NewPrepareStatementError(ctx, &daoImpl, getSourceByTenantId, err)
+	}
+	return &daoImpl, nil
+}
+
+func (dao *sourceDaoSqlx) NameForError() string {
+	return dao.name
+}
+
+func init() {
+	dao.GetSourceDao = getSourceDao
+}
+
+func (dao *sourceDaoSqlx) Create(ctx context.Context, source *models.Source) (*models.Source, error) {
```

Comment:
No idea why linter doesn't complain here, but please rename `dao` to something else so we do not shadow the package `dao`.